main.js

```diff
@@ -0,0 +1,150 @@
+//操作するHTMLを取得
+const titleLogo = document.getElementById('titleLogo');
+const text = document.getElementById('text');
+const btn = document.getElementById('btn');
+const Api = 'https://opentdb.com/api.php?amount=10';
+//問題番号を管理する変数
+let quizNumber = 0;
+//正解数をカウントする変数
+let count = 0;
+//回答欄の親要素を作成
+const container = document.createElement('div');
+
+//ボタンを押したときにAPIを取得する
+btn.addEventListener('click', function () {
+  //API取得中に表示するテキスト
+  titleLogo.textContent = '取得中';
+  text.textContent = '少々お待ちください';
+  btn.style.display = 'none';
+  chachApi(text)
+});
+
+//JSONデータを取得するAPI
+const chachApi = function (text) {
+  fetch(Api)
+    .then(response => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        throw new Error();
+      }
+    })
+    .then(data => {
+      //問題の順番を管理・表示する変数
+      Quiz(data);
+    })
+    .catch(error => {
+      console.log(error);
+    });
+}
```

Comment:
- `await` を使って書く方がイマドキの書き方ですし、他言語との差異も少ないのでおススメです
- `chach` がおそらく Typo かなと思うのですが…ややわかりにくい命名だったので、名前を考え直してほしいです
  - コードの例は私の提案を書いたまでなので、無理に採用いただかなくてもいいです
- 変数 `text` を利用していないようだったので、消した方がよいかと思いました

```suggestion
const fetchQuizData = async () => {
  try {
    const response = await fetch(Api);
    const json = await response.json();
    Quiz(json);
  } catch (e) {
    console.error(e);
  }
}
```

Reply:
ありがとうございます。
awaitを取り入れました。asyncとセットの認識で使う形でよろしかったでしょうか？
ざっくりですが、これらの2つに関して下記のような理解で使っています。
async：非同期関数を定義する、アロー関数は利用できない 
await:指定した関数の結果が返されるまで全体の処理を待機する

また、上記に関しては下記の記事を参考にしました。
https://rightcode.co.jp/blog/information-technology/javascript-async-await

Reply:
>async：非同期関数を定義する、アロー関数は利用できない
await:指定した関数の結果が返されるまで全体の処理を待機する

両方とも一部認識間違っております。
asyncは非同期関数を定義する時に付けますが、アロー関数でも定義できます。
上記にmiyazakiさんが記載してくれているコードはアロー関数で定義されておりますね。

awaitは同期的に記述できる書き方なのですが、「全体の処理を待機する」という認識が間違っていて、awaitの理解というよりも非同期処理の理解に誤りがあります。
非同期処理とは、
「関数が呼び出されたとき、戻り値として本来渡したい結果を返すのではなく、一度関数としては終了し（＝呼び出し元に戻る）、後で『本来渡したかった値』を返せる状態になったときに、呼び出し元にその値を通知する」という仕組みです。
なので、全体の処理が止まる訳ではありません。

こちらの記事が良くまとまっていて、コメントに一部抜粋させていただきました。
https://qiita.com/kiyodori/items/da434d169755cbb20447

まだnode.jsに関することは分からないかもしれませんが、ざっと目を通しておくだけでも理解が深まるかと思います。

>chachに関してはMDNみたらエラーを取得する際に使う仕様のようなのでそのままにしました。

これに関してはご指摘内容に少し誤解をなさっていて、miyazakiさんがご指摘したのは、chachApiという命名かと思います。
chachではなくcatchのtypoではないか、というご指摘ですね。
エラーハンドリングでcatchを使用するのは正しいです。

Reply:
レビューありがとうございます。
  記事も拝見させて頂きました。概要はおおよそ理解できましたが、まだ使いこなえるレベルに至っていない感じがするのでVueの学習と並行してJavaScriptの非同期処理や他の弱そうな部分のキャッチアップも継続したいと思います！
  ！